devito/propagator.py

```diff
@@ -254,12 +255,33 @@ def convert_equality_to_cgen(self, equality):
             return equality
         else:
             s_lhs = ccode(self.time_substitutions(equality.lhs).xreplace(self._var_map))
-            s_rhs = ccode(self.time_substitutions(equality.rhs).xreplace(self._var_map))
+            s_rhs = self.time_substitutions(equality.rhs).xreplace(self._var_map)
+
+            self.sub_stencils.append(s_rhs)  # appending substituted stencil,which is used to determine alignment pragma
+
+            s_rhs = ccode(s_rhs)
             if self.dtype is np.float32:
                 s_rhs = str(s_rhs).replace("pow", "powf")
                 s_rhs = str(s_rhs).replace("fabs", "fabsf")
             return cgen.Assign(s_lhs, s_rhs)
 
+    def set_aligned_pragma(self, stencils, factorized, loop_counters, time_steppers):
+        """
+        Sets the alignment for the pragma.
+        :param stencils: List of stencils.
+        :param factorized:  dict of factorized elements
+        :param loop_counters: list of loop counter symbols
+        :param time_steppers: list of time stepper symbols
+        """
+
+        array_names = set()
+        for item in flatten([stencil.free_symbols for stencil in stencils]):
+            if str(item) not in factorized and item not in loop_counters + time_steppers:
+                array_names.add(item)
+
+        pragma = "omp simd aligned(%s:64)" % ", ".join([str(i) for i in array_names])
```

Comment:
Good catch. I think we talked about this before. Such constants should be determined programmatically using sysctlbyname: http://www.manpagez.com/man/3/sysctlbyname/
In the case of cache line size you are looking for hw.cachelinesize

However, and I believe @FabioLuporini raised this previously as well, we might instead want to align to page size in which case we should use hw.pagesize so we can reduce TBL misses. The amount of wasted memory is still small.